src/payload.rs

```diff
@@ -69,6 +73,15 @@ impl Payload {
     pub fn insert(&mut self, key: impl ToString, val: impl Into<Value>) {
         self.0.insert(key.to_string(), val.into());
     }
+
+    /// Deserializes the payload directly into `T`. This requires T to implement `serde::Deserialize`.
+    /// Returns an error if `T` and the payload have a different structure.
```

Comment:
Could you add a simple example in this comment?

Maybe in a similar fashion as we do with other functions: https://docs.rs/qdrant-client/latest/qdrant_client/struct.Qdrant.html#method.set_payload